apps/app/src/app/knowledge-box/knowledge-box-profile/knowledge-box-profile.component.ts

```diff
@@ -52,20 +54,23 @@ export class KnowledgeBoxProfileComponent implements OnInit, OnDestroy {
   }
 
   initKbForm(): void {
+    this.kbForm.get('uid')?.patchValue(this.kb?.id);
+    this.kbForm.get('slug')?.patchValue(this.kb?.slug);
     this.kbForm.get('title')?.patchValue(this.kb?.title);
     this.kbForm.get('description')?.patchValue(this.kb?.description);
   }
 
   saveKb(): void {
     if (this.kbForm.invalid) return;
+    const newSlug = STFUtils.generateSlug(this.kbForm.value.slug);
     const data: Partial<KnowledgeBox> = {
       title: this.kbForm.value.title,
       description: this.kbForm.value.description,
-      // anonymization: this.useAnonymization ? 'anonymization' : '',
+      slug: newSlug,
     };
     (this.kb as WritableKnowledgeBox)
       .modify(data)
-      .pipe(concatMap(() => this.sdk.nuclia.db.getKnowledgeBox(this.account!.slug, this.kb!.slug!)))
+      .pipe(concatMap(() => this.sdk.nuclia.db.getKnowledgeBox(this.account!.slug, newSlug)))
       .subscribe((kb) => {
         this.stateService.setStash(kb);
```

Comment:
don't we need to re-route? (as the current route contain the old slug, it can break navigation, no?)

Reply:
The current route contain the old slug yes, but as soon as you navigate somewhere else the slug is automatically updated without requiring re-routing.